### DISCUSSION TITLE
Panel with direction to waypoint

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#217] [Q&A] [answered]
Currently, the compass shows us the directions of the world and its long pressing allows you to change the map rotation mode. There were already ideas to give it a short press function (like mark the "Wow" or take a screenshot).

Or maybe we would like the icon to indicate the destination (this is the choice I have in other navigation)? It could be an option or a combo icon so you don't have extra settings. E.g. something like this:

![Screenshot_20230615_140723_Cruiser](https://github.com/devemux86/cruiser/assets/122647782/7425b81a-9e29-4480-afb7-2748262b9fbb)

**Comment by devemux86** [comment, marked as answer]
@Ciutnik thanks for the suggestion.

This is already available in navigation: when off route, there is a direction arrow with distance to next waypoint.

**Comment by Ciutnik** [comment]
> What other navigators have it so we can see their screens?

In an app that works locally in my country (NaviExpert – brilliantly bypasses traffic jams thanks to the very large user community) you can choose what the small icon in the upper right corner should indicate: compass or bearing to your destination. 

<img src="https://github.com/devemux86/cruiser/assets/122647782/fe036619-dc24-4073-b762-b4636aa83fdc" width="49%"/>  <img src="https://github.com/devemux86/cruiser/assets/122647782/33a7ff7d-1a71-4cb2-b12d-dbd9f79f68c3" width="49%"/>

> Also it could be destination or next via point.

I was wondering about that too. Probably too much information if you put both arrows. I guess the destination should have a higher priority (?).

**Reply by devemux86** [reply]
> you can choose what the small icon in the upper right corner should indicate: compass or bearing to your destination

Very nice implementation. :+1: 

> I guess the destination should have a higher priority (?).

Or we could do it like the navigation panels?
Let the user choose what they want to see: next via point or destination.

**Reply by devemux86** [reply]
Also the direction to a selected next waypoint could be displayed in:

- **Distance panel** (top right)
(now it has a fixed icon)

- **Next waypoint panel** (top horizontal)
(now it does not have an icon, but it is an extra panel)

**Reply by WolfgangZ42** [reply]
> Distance panel (top right) - (now it has a fixed icon)

It's not a fixed icon ... ![2023-06-15 20_13_47-2023-06-15 20 12 54 png - IrfanView (Zoom_ 470 x 939)](https://github.com/devemux86/cruiser/assets/121806238/45693214-288f-4094-b0eb-d95cb185f2f2) for destination and ![2023-06-15 20_15_00-2023-06-15 20 12 48 png - IrfanView (Zoom_ 470 x 939)](https://github.com/devemux86/cruiser/assets/121806238/1a2f0366-1a5e-4209-acb4-0be9a34d6061) for next waypoint.
And for my opinion this is important enough ...

**Comment by Ciutnik** [comment]
@devemux86 
> Let the user choose what they want to see: next via point or destination.

A short press could toggle between the compass (like now) a yellow bearing arrow on the via-point and a red one on the destination waypoint? The colors would be consistent with the symbols on the route.

@WolfgangZ42 
You're right, although the bearing icon on the distance panel is a nice idea, but we are already used to the current symbols, which are unambiguous and important.

**Comment by WolfgangZ42** [comment]
Unfortunately there's no search function in github, so I selected this discussion ...

A long tap on the compass sign shows the following choices in Cruiser:
<img width="400" src="https://github.com/devemux86/cruiser/assets/121806238/5fe553ba-5ba3-44e2-b701-d8fe3c701bb6"/>

Every now and then (when using Cruiser in the car) I miss an option that was there in K2:
<img width="400" src="https://github.com/devemux86/cruiser/assets/121806238/4cec9b53-bdfb-4ea2-9047-67175942d43c"/>

I know there is the overview button:
<img width="400" src="https://github.com/devemux86/cruiser/assets/121806238/8ed698dd-00f0-4dd2-8870-468e1ce3a657"/>

**But this only shows the whole route temporarily.** While in K2 the **permanent** overview of the **remaining distance** was interesting. And at the target the zoom lets you see even the smallest side road and house numbers. 
Do you see a chance to get "Route Overview" back? 🤔

**Reply by devemux86** [reply]
> there's no search function in github

Search is on the top of the Discussions:

![search](https://github.com/devemux86/cruiser/assets/3484020/8a2378f1-b634-4041-9407-5610f1cd4f22)

**Reply by devemux86** [reply]
@WolfgangZ42 thanks for the detailed suggestion!

>  Do you see a chance to get "Route Overview" back?

Route overview of the remaining route?
It can return in Cruiser again. :slightly_smiling_face: 

Should we have it in compass or in the actions?
Although the actions better be an even number for symmetry.

**Reply by Ciutnik** [reply]
@WolfgangZ42 
> But this only shows the whole route temporarily. While in K2 the permanent overview of the remaining distance was interesting.

For me, **because temporary** it is very useful. In K2, I didn't like having to manually return to the regular view. That's why I usually used zoom-out multiple times.

So maybe put this as an option in the auto-zoom settings? In addition to the numerical value "Maximum auto zoom" add "To the end of the route"?

**Reply by WolfgangZ42** [reply]
Okay, if the temporary approach is useful for others, perhaps we could have both? 😏 
Keep the overview buttom in the action menu and return the "remaining route" into map orientation (compass sign).

**Reply by devemux86** [reply]
The idea to select the route overview behavior with a setting is interesting.

Note: it is not only the appearance of the full route or the remaining route.
It is also the timer, existing function of route overview is with a timer.

**Reply by Ciutnik** [reply]
My proposal seems nice because it does not conflict with the "Distance" or even "Speed + Distance" setting (~~of course without the reverse variants~~). I like to see a large area of ​​the map, and with short trips of 50-70 km, I would like to see its end point.

Edit: basically with "reverse" variants this can work well too!

**Reply by devemux86** [reply]
@Ciutnik I do not see how it is related to any auto zoom variant, as it is something completely separate.

Overview of remaining route is a dynamic zoom that keeps in screen the bounds of the remaining route.
As we progress, it's expected to zoom in more and more. At least this is how it worked in the past.

Is this what we are discussing to return in compass (or elsewhere)?

**Reply by WolfgangZ42** [reply]
> Overview of remaining route is a dynamic zoom that keeps in screen the bounds of the remaining route.

Yes - this is it. And for my opinion it's more a map orientation as a zoom variant. The latter is a more static definition whereby I can switch the map orientation dynamically.

**Reply by Ciutnik** [reply]
@devemux86 
> I do not see how it is related to any auto zoom variant

Now, after reaching "Maximum speed", the zoom value reaches the value set in "Maximum auto zoom". I mean, instead of a numerical zoom value, you can choose something like "Show to the end of the route" (or simply "Overview"), which will constantly change the zoom to show the remaining route. Which is what we know now as dynamic "Overview". 

In other words: "Overview" turns on automatically after reaching "Maximum speed" 🙂

**Reply by devemux86** [reply]
@Ciutnik thanks, now I understand!

Indeed, this seems like another interesting idea.

**Comment by DolomitesTom** [comment]
what about my former proposal to make different zoom settings around the actual GPS position till the end of the route? 1x tap compass = 20% of the route, 2x tap = 40%, 3x tap = 50%, 4x tap = 80% for example. 100% is already available via action menu.

**Reply by devemux86** [reply]
@DolomitesTom we have a related topic here:
- https://github.com/devemux86/cruiser/discussions/94
